processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -23,7 +37,101 @@ func (dp Histogram) Add(in Histogram) Histogram {
 	panic("todo")
 }
 
-// nolint
 func (dp ExpHistogram) Add(in ExpHistogram) ExpHistogram {
-	panic("todo")
+	switch {
+	case dp.Timestamp() >= in.Timestamp():
+		fail("out of order")
+	case dp.Scale() != in.Scale():
+		fail("scale changed")
+	case dp.ZeroCount() != in.ZeroCount():
```

Comment:
I believe the zero count can be added between the histograms; if the zero thresholds are not the same, they should be able to be combined by taking the larger threshold between the histograms and the bucket(s) that fall into that larger threshold from the histogram with the smaller threshold can be merged. There's some info on this here: https://opentelemetry.io/docs/specs/otel/metrics/data-model/#zero-count-and-zero-threshold